main.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"log"
+	"os"
+
+	"github.com/go-telegram-bot-api/telegram-bot-api"
+	"github.com/openai/openai-go/v1"
+)
+
+func main() {
+	// Initialize OpenAI API client
+	client := openai.NewClient(os.Getenv("OPENAI_API_KEY"))
+
+	// Initialize Telegram bot API
+	bot, err := tgbotapi.NewBotAPI(os.Getenv("TELEGRAM_BOT_TOKEN"))
+	if err != nil {
+		log.Panic(err)
+	}
+
+	// Set bot to receive updates from all messages
+	u := tgbotapi.NewUpdate(0)
+	u.Timeout = 60
+
+	updates, err := bot.GetUpdatesChan(u)
+
+	// Process incoming messages
+	for update := range updates {
+		if update.Message == nil {
+			continue
+		}
+
+		// Process user input in natural language
+		if update.Message.From.UserName != bot.Self.UserName {
+			response, err := client.Completions(
+				openai.CompletionRequest{
```

Comment:
depending on user's request, the assistant can consider calling the agent to get more information. For example, if a user asks for some recent news, the assistant can first call agent to search for it on the internet, and then, it can absorb what the agent gives and then answer the user's request.

This can cycle a few rounds until the assistant is confident to give the final answer.